src/sql-parser/src/ast/defs/ddl.rs

```diff
@@ -264,6 +271,20 @@ impl AstDisplay for Connector {
                 f.write_node(&display::escape_single_quote_string(path));
                 f.write_str("'");
             }
+            Connector::S3 {
+                bucket,
+                objects_pattern,
+                order,
+            } => {
+                f.write_str("S3 BUCKET '");
+                f.write_str(&display::escape_single_quote_string(bucket));
+                f.write_str("' OBJECTS '");
+                f.write_str(&display::escape_single_quote_string(objects_pattern));
+                f.write_str("'");
+                if *order {
+                    f.write_str(" ORDER BY OBJECT KEY");
```

Comment:
Not sure if we should be repurposing bits of SQL syntax that have different meaning in other contexts (`ORDER BY`) for something new here. Should probably ask Nikhil what he thinks

Reply:
It was proposed in the "syntax" section of https://github.com/MaterializeInc/materialize/issues/4914 , which I suppose we should CC @benesch to take a closer look at.

Reply:
I spoke with nikhil offline and he has no objections to this syntax, but we may consider using something different like `SERIAL` before we remove the `--experimental` requirement.